pkg/partmgr/partmgr.go

```diff
@@ -261,27 +266,35 @@ func (p *PartitionManager) ReadAndUpdateSchema() (err error) {
 		p.schemaMtimeNanosecs = mtimeNsecs
 
 		metricReporter.WithTimer("ReadAndUpdateSchemaTimer", func() {
-			resp, err := p.container.GetObjectSync(&v3io.GetObjectInput{Path: fullPath})
-			if err != nil {
-				err = errors.Wrapf(err, "Failed to read schema at path '%s'.", fullPath)
+			resp, innerError := p.container.GetObjectSync(&v3io.GetObjectInput{Path: fullPath})
+			if innerError != nil {
+				err = errors.Wrapf(innerError, "Failed to read schema at path '%s'.", fullPath)
```

Comment:
Shouldn't we return here? Otherwise, it will try to unmarshal an empty response and the error will be overwritten, then it will try to update the schema and either overwrite the error again, or corrupt the schema.